diff --git a/src/helpers/ESP32Board.h b/src/helpers/ESP32Board.h
index bade3e8..294a381 100644
--- a/src/helpers/ESP32Board.h
+++ b/src/helpers/ESP32Board.h
@@ -57,7 +57,7 @@ public:
   }
 
   void enterLightSleep(uint32_t secs) {
-#if defined(CONFIG_IDF_TARGET_ESP32S3) && defined(P_LORA_DIO_1) // Supported ESP32 variants
+#if defined(P_LORA_DIO_1) && (defined(CONFIG_IDF_TARGET_ESP32S3) || defined(CONFIG_IDF_TARGET_ESP32C6)) // Supported ESP32 variants
     if (rtc_gpio_is_valid_gpio((gpio_num_t)P_LORA_DIO_1)) { // Only enter sleep mode if P_LORA_DIO_1 is RTC pin
       esp_sleep_pd_config(ESP_PD_DOMAIN_RTC_PERIPH, ESP_PD_OPTION_ON);
       esp_sleep_enable_ext1_wakeup((1L << P_LORA_DIO_1), ESP_EXT1_WAKEUP_ANY_HIGH); // To wake up when receiving a LoRa packet
@@ -77,6 +77,19 @@ public:
     }
   }
 
+  void sleepMs(uint32_t ms) {
+    if (!inhibit_sleep) {
+#if defined(P_LORA_DIO_1) && (defined(CONFIG_IDF_TARGET_ESP32S3) || defined(CONFIG_IDF_TARGET_ESP32C6))
+      if (rtc_gpio_is_valid_gpio((gpio_num_t)P_LORA_DIO_1)) {
+        esp_sleep_pd_config(ESP_PD_DOMAIN_RTC_PERIPH, ESP_PD_OPTION_ON);
+        esp_sleep_enable_ext1_wakeup((1L << P_LORA_DIO_1), ESP_EXT1_WAKEUP_ANY_HIGH);
+        esp_sleep_enable_timer_wakeup((uint64_t)ms * 1000ULL);
+        esp_light_sleep_start();
+      }
+#endif
+    }
+  }
+
   uint8_t getStartupReason() const override { return startup_reason; }
 
 #if defined(P_LORA_TX_LED)
